Netkan/Sources/Github/GithubApi.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Linq;
+using System.Net;
+using log4net;
+using Newtonsoft.Json.Linq;
+
+// We could use OctoKit for this, but since we're only pinging the
+// release API, I'm happy enough without yet another dependency.
+
+namespace CKAN.NetKAN.Sources.Github
+{
+    internal sealed class GithubApi : IGithubApi
+    {
+        private static readonly ILog Log = LogManager.GetLogger(typeof(GithubApi));
+        private static readonly Uri ApiBase = new Uri("https://api.github.com/");
+
+        private readonly string _oauthToken;
+
+        public GithubApi(string oauthToken = null)
+        {
+            _oauthToken = oauthToken;
+        }
+
+        public GithubRelease GetLatestRelease(GithubRef reference)
+        {
+            var json = Call(string.Format("repos/{0}/releases", reference.Repository));
+            Log.Debug("Parsing JSON...");
+            var releases = JArray.Parse(json);
+
+            // Finding the most recent *stable* release means filtering
+            // out on pre-releases.
+
+            foreach (var release in releases)
+            {
+                // First, check for prerelease status...
+                if (reference.UsePrelease == (bool)release["prerelease"])
```

Comment:
Oh! This is checking to see if the release status is the same as what we're expecting. In other words:

1. If we're run with `--prerelease`, we'll only ever index a pre-release, even if a more recent stable release exists.
1. If we're run without `--prerelease`, we'll only ever index a stable release.

I guess nobody's complained about that so far? (In a strong multi-repo model I'm using to seeing *all* releases in the "dev" repositories, but I doubt this really has any impact on how our users work.)